challenges.py

```diff
@@ -28,17 +26,15 @@
 for user in users:
     for challenge in user["challenges"]:
         challengeTitle = challenge["title"]
-        challengeWinnersPerChallenge.setdefault(challengeTitle, set())
+        winners = challengeWinnersPerChallenge.setdefault(challengeTitle, set())
         if challenge["points"] >= challenge["maxPoints"]:
-            winners = challengeWinnersPerChallenge.get(challengeTitle, set())
             username = user["username"]
             if username not in excluded:
                 winners.add(user["username"])
-            challengeWinnersPerChallenge.setdefault(challengeTitle, winners)
```

Comment:
Not needed because we're mutating the existing set which is already in the dictionary - so we don't need to set it afterwards, it's still the same object - same like e.g. here:

```python3
d = {"values": []}
values = d["values"]
values.append(42)
print(d["values"])  # [42]
```

which is equivalent to:

```python3
d = {"values": []}
d["values"].append(42)
print(d["values"])  # [42]
```